host/windows/syscall.c

```diff
@@ -324,6 +345,617 @@ __declspec(noreturn) static void _panic(
+        goto done;
+    }
+
+    psid[0] = OwnerInfo->Owner;
+
+    // Call GetTokenInformation to get the buffer size.
+    if(!GetTokenInformation(hToken, TokenPrimaryGroup, NULL, 0, &dwSize))
+    {
+        dwRes = GetLastError();
+        if(dwRes != ERROR_INSUFFICIENT_BUFFER) {
+            _set_errno(GetLastError());
+            goto done;
+        }
+    }
+
+    // Allocate the buffer.
+    GroupInfo = (TOKEN_PRIMARY_GROUP *) calloc(dwSize, sizeof(char));
+
+    // Call GetTokenInformation again to get the group information.
+    if(!GetTokenInformation(hToken, TokenPrimaryGroup, GroupInfo,
+                            dwSize, &dwSize))
+    {
+        _set_errno(GetLastError());
+        goto done;
+    }
+
+    psid[1] = GroupInfo->PrimaryGroup;
+
+    // Create a SID for the Everyone group.
+    if (!AllocateAndInitializeSid(&SIDAuthWorld, 1,
+                     SECURITY_WORLD_RID,
+                     0,
+                     0, 0, 0, 0, 0, 0,
+                     &psid[2]))
+    {
+        _set_errno(GetLastError());
+        goto done;
+    }
+
+    EXPLICIT_ACCESS ea[NUM_ACES];
+    ZeroMemory(&ea, NUM_ACES * sizeof(EXPLICIT_ACCESS));
+
+    // Track the number of effective ea, to avoid blank permission ea.
+    int num_aces = 0;
+
+    for (int i = 0; i < 3; i++)
+    {
+        // Set access to each ea. According to ACE canoical rule,
+        // deny access must proceed before set access.
+        if (i == 1 && denies[i])
+        {
+            ea[num_aces].grfAccessPermissions = denies[i];
+            ea[num_aces].grfAccessMode = DENY_ACCESS;
+            ea[num_aces].grfInheritance = NO_INHERITANCE;
+            ea[num_aces].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+            ea[num_aces].Trustee.TrusteeType = TRUSTEE_IS_WELL_KNOWN_GROUP;
+            ea[num_aces].Trustee.ptstrName = (LPTSTR) psid[i];
+
+            num_aces++;
+        }
+
+        if (grants[i])
+        {
+            ea[num_aces].grfAccessPermissions = grants[i];
+            ea[num_aces].grfAccessMode = SET_ACCESS;
+            ea[num_aces].grfInheritance = NO_INHERITANCE;
+            ea[num_aces].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+            ea[num_aces].Trustee.TrusteeType = TRUSTEE_IS_WELL_KNOWN_GROUP;
+            ea[num_aces].Trustee.ptstrName = (LPTSTR) psid[i];
+
+            num_aces++;
+        }
+
+    }
+
+    // Write the effective ea to DACL.
+    if (ERROR_SUCCESS != SetEntriesInAcl(num_aces,
+                                         ea,
+                                         NULL,
+                                         &pACL))
+    {
+        _set_errno(GetLastError());
+        goto done;
+    }
+
+    // Deal with directory here.
+    // Compared to file, dir has much more aces as for CREATOR OWNER, GROUP
+    // It is much easy to let Windows finish the creation then apply the PACL.
+    if (dwDesiredAccess == FILE_DIRECTORY_FILE)
+    {
+        if (!CreateDirectoryA((LPTSTR) FileName, NULL))
```

Comment:
```suggestion
        if (!CreateDirectoryA((PCSTR) FileName, NULL))
```

Explanation: T types would be used without the A, but the A pins it to the PCSTR type

Reply:
Fixed. I even didn't notice the difference, thanks.